traitlets/traitlets.py

```diff
@@ -499,18 +499,33 @@ def init_default_value(self, obj):
         obj._trait_values[self.name] = value
         return value
 
+    def get_env_value(self):
+        """
+        Gets the value of any environment-variable named as `env` metadata.
+
+        :return:
+            the textual value of the environment variable, or None, meaning
+            that no `envvar` metadata is defined.
+        """
+        env_var = self.metadata.get('envvar')
+        return env_var and os.environ.get(env_var)
```

Comment:
(oups, was to ambitious, and had forgotten the purpose of my here)

The point above was NOT to allow for `None`; it was for empty-string(`''`) to pass through to the trait - that's a very common case. 
`None` is impossible to signify from environment; that is why I used it as a sentinel to mean "no-env" var exists at all.